Bigtable/src/Instance.php

```diff
@@ -0,0 +1,362 @@
+        $labels = $this->pluck('labels', $options, false) ?: [];
+        $type = $this->pluck('type', $options, false);
+
+        $clustersArray = [];
+        foreach ($clusterMetadataList as $value) {
+            if (!isset($value['clusterId'])) {
+                throw new \InvalidArgumentException('Cluster id must be set.');
+            }
+            $this->validate($value['clusterId'], 'cluster');
+            $clusterId = $value['clusterId'];
+            if (!isset($value['locationId'])) {
+                throw new \InvalidArgumentException('Location id must be set.');
+            }
+            $this->validate($value['locationId'], 'location');
+            $locationId = $value['locationId'];
+            $value['location'] = InstanceAdminClient::locationName($this->projectId, $locationId);
+            if ($type === Instance_Type::DEVELOPMENT) {
+                unset($value['serveNodes']);
+            } elseif ($type === Instance_Type::PRODUCTION
+                        && (!isset($value['serveNodes']) || $value['serveNodes'] <= 0)) {
+                throw new \InvalidArgumentException('When creating Production instance, serveNodes must be > 0');
+            }
+            // `$clustersArray` must be keyed by the cluster ID.
+            $clustersArray[$clusterId] = $value;
+        }
+        $operation = $this->connection->createInstance([
+            'parent' => $projectName,
+            'instanceId' => $this->id,
+            'instance' => [
+                'displayName' => $displayName,
+                'type' => $type,
+                'labels' => $labels
+            ],
+            'clusters' => $clustersArray
+        ] + $options);
+
+        return $this->resumeOperation($operation['name'], $operation);
+    }
+
+    /**
+     * This method returns an associative array containing data required for
+     * configuring Bigtable instance clusters. The resulting array will contain
+     * 'clusterId', 'locationId`, and depending on how it is called,
+     * `storageType` and `serveNodes`.
+     *
+     * Example:
+     * ```
+     * $cluster = Instance::buildClusterMetadata('my-cluster', 'us-east1-b');
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @param string $clusterId The cluster ID
+     *        e.g., just `cluster-id` rather than `projects/project-id/instances/instance-id/clusters/cluster-id`.
+     * @param string $locationId The location ID
+     *        e.g., just `us-east1-b` rather than `projects/project-id/locations/us-east1-b`.
+     * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
+     *        `Google\Cloud\Bigtable\Admin\V2\StorageType::SSD`,
+     *        `Google\Cloud\Bigtable\Admin\V2\StorageType::HDD`.
+     *        **Defaults to** @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.StorageType StorageType
```

Comment:
Could we follow the suggestion outlined [here](https://github.com/GoogleCloudPlatform/google-cloud-php/pull/1158#discussion_r202142860) please? :) (this applies to other cases where we mention the default server value as well)

```php
/**
 * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
 *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`,
 *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_HDD`.
 *        **Defaults to** `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`.
 */
```

Reply:
had discussion with @sduskis and we decided to point to api specification for defaults rather than having it here.

Reply:
Okay, that seems fine in general. It is worth noting the link here doesn't say anything about the default value, though. If we do this, to help make this a little more clear could we please reword it just a bit and use a markdown link?

```php
/**
 * @param int $storageType The storage media type for persisting Bigtable data. Possible values include
 *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_SSD`,
 *        `Google\Cloud\Bigtable\Instance::STORAGE_TYPE_HDD`.
 *        For the default value, please see [the upstream documentation](https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#google.bigtable.admin.v2.StorageType).
 */
```

Reply:
Fixed